appscale/tools/agents/ec2_agent.py

```diff
@@ -147,26 +156,33 @@ def configure_instance_security(self, parameters):
         "value, or erase it to have one automatically generated for you." \
         .format(keyname))
 
-    security_groups = conn.get_all_security_groups()
-    for security_group in security_groups:
-      if security_group.name == group:
-        self.handle_failure("Security group {0} is already registered. Please" \
-          " change the 'group' specified in your AppScalefile to a different " \
-          "value, or erase it to have one automatically generated for you." \
-          .format(group))
+    try:
+      self.get_security_group_by_name(parameters, group)
+      self.handle_failure("Security group {0} is already registered. Please "
```

Comment:
This should be in an `else` clause after the except. Otherwise, the `AgentRuntimeException` that `handle_failure` raises will be caught by the except.

From a readability standpoint, it also makes it clearer that the call is supposed to fail (a success being a more exceptional event).

Reply:
Can I mark this one as resolved?